### PR TITLE
pda does not require a signer in metadata

### PR DESCRIPTION
**Pull request by seanyoung**
If we have a seed, we don't need a signer.

### CODE REVIEW

File: src/abi/anchor.rs [hunk 1]
```diff
@@ -157,7 +157,9 @@ fn idl_instructions(
                 vec![IdlAccountItem::IdlAccount(IdlAccount {
                     name: "dataAccount".to_string(),
                     is_mut: true,
-                    is_signer: func.has_payer_annotation(),
+                    /// With @payer annotation, the account is created on-chain; however, if a @seed
+                    /// is provided, then th
```
Comment (LucasSte) [inline src/abi/anchor.rs:161]: This comment is incomplete.

**Review by LucasSte — CHANGES_REQUESTED** [review]

File: src/abi/anchor.rs (hunk 1 above)
Comment (seanyoung) [reply src/abi/anchor.rs]: I don't know what happened, must have been distracted half-way through typing that! Fixed

File: src/abi/anchor.rs [hunk 2]
```diff
@@ -157,7 +157,14 @@ fn idl_instructions(
                 vec![IdlAccountItem::IdlAccount(IdlAccount {
                     name: "dataAccount".to_string(),
                     is_mut: true,
-                    is_signer: func.has_payer_annotation(),
+                    /// With @payer annotation, the account is created on-chain and needs a signer. The client
+                    /// provides an address which does not exist yet, so SystemProgram.CreateAccount is called
```
Comment (LucasSte) [inline src/abi/anchor.rs:161]:
```suggestion
                    /// provides an address that does not exist yet, so SystemProgram.CreateAccount is called
```
This is a restrictive clause.

File: src/abi/anchor.rs [hunk 3]
```diff
@@ -157,7 +157,14 @@ fn idl_instructions(
...
+                    /// on-chain.
+                    ///
+                    /// However, if a @seed is provided, then the program can sign for the account
```
Comment (LucasSte) [inline src/abi/anchor.rs:164]:
```suggestion
                    /// However, if a @seed is provided, the program can sign for the account
```
"Omit needless words"

File: src/abi/anchor.rs [hunk 4]
```diff
@@ -157,7 +157,14 @@ fn idl_instructions(
...
+                    /// using the seed using program derived address (pda) when SystemProgram.CreateAccount is called,
```
Comment (LucasSte) [inline src/abi/anchor.rs:165]: These two `using` here seem a bit awkward. Is that what you wanted to say?

**Review by xermicus — APPROVED** [review]

File: src/abi/anchor.rs (hunk 4 above)
Comment (seanyoung) [reply src/abi/anchor.rs]: I agree, improved the wording

File: src/abi/anchor.rs [hunk 5]
```diff
@@ -157,7 +157,14 @@ fn idl_instructions(
                 vec![IdlAccountItem::IdlAccount(IdlAccount {
                     name: "dataAccount".to_string(),
                     is_mut: true,
-                    is_signer: func.has_payer_annotation(),
+                    /// With a @payer annotation, the account is created on-chain and needs a signer. The client
+                    /// provides an address that does not exist yet, so SystemProgram.CreateAccount is called
+                    /// on-chain.
+                    ///
+                    /// However, if a @seed is also provided, then program can sign for the account
```
Comment (LucasSte) [inline src/abi/anchor.rs:164]:
```suggestion
                    /// However, if a @seed is also provided, the program can sign for the account
```

**Review by LucasSte — APPROVED** [review]

**Pull request merged**